src/Logging/Dispatcher.cpp

```diff
@@ -25,16 +26,34 @@ void Dispatcher::run() {
         }   
     }
 
-    // call log on loggers and reset logThisLoop bool
+    // check max publish sizes, publish if DataQueue buffer is close to full, update max publish sizes for each logger
     if (_logThisLoop) {
-        _dataQ->wrapStart();
+        bool firstPass = true;
         for (uint16_t i = 0; i < _numLoggers; i++) {
+            if (_dataQ->getDataSize() + _maxPublishSizes[i] + 2 > _dataQ->getBufferSize() && _loggers[i]->executeThisLoop())
+                _dataQ->publish("BQIngestion", PRIVATE, WITH_ACK);
```

Comment:
Can we define the event ("BQIngestion" in this case) as a constant or pass it into the constructor for this/DataQueue? Whatever makes more sense to you.

Reply:
might make the most sense to pass it into the constructor, that way we could have a different implementation depending on the current vehicle